src/NUnitEngine/nunit.engine/Runners/MasterTestRunner.cs

```diff
@@ -374,6 +377,24 @@ private TestEngineResult RunTests(ITestEventListener listener, TestFilter filter
             return result;
         }
 
+        private AsyncTestEngineResult RunTestsAsync(ITestEventListener listener, TestFilter filter)
```

Comment:
This method is copied from AbstractTestRunner. In 3.4.1, MasterTestRunner inherited from AbstractTestRunner and the method was called. We eliminated that in separating MasterTestRunner from the internal engine runners. Since we don't call RunAsync in the console or VS adapter, the error went unnoticed until we updated the Gui to use the 3.5 engine.